src/FamilyHubs.Referral.Core/Commands/ProcessGridEventCommand.cs

```diff
@@ -197,45 +197,63 @@ private T Deserialize<T>(string? json)
 
     private async Task ProcessUserAccount(UserAccountDto userAccountDto, CancellationToken cancellationToken)
     {
-        //Process Custom Event
-        _logger.LogInformation("Processing User Account Event Grid event message");
+        try
+        {
+            //Process Custom Event
+            _logger.LogInformation("Processing User Account Event Grid event message");
 
-        _logger.LogInformation($"Creating User Account for Processing Events: {userAccountDto.Name}-{userAccountDto.EmailAddress}");
+            _logger.LogInformation($"Creating User Account for Processing Events: {userAccountDto.Name}-{userAccountDto.EmailAddress}");
 
-        var user = _context.UserAccounts.FirstOrDefault(x => x.Id == userAccountDto.Id);
-        if (user != null)
-        {
-            UpdateUserAccountCommand updateUserAccountCommand = new UpdateUserAccountCommand(userAccountDto.Id, userAccountDto);
-            await _mediator.Send(updateUserAccountCommand, cancellationToken);
+            var user = _context.UserAccounts.FirstOrDefault(x => x.Id == userAccountDto.Id);
+            if (user != null)
+            {
+                UpdateUserAccountCommand updateUserAccountCommand = new UpdateUserAccountCommand(userAccountDto.Id, userAccountDto);
+                await _mediator.Send(updateUserAccountCommand, cancellationToken);
+            }
+            else
+            {
+                //Will have throw a NotFoundException so need to add
+                CreateUserAccountCommand command = new(userAccountDto);
+                await _mediator.Send(command, cancellationToken);
+            }
         }
-        else
+        catch (Exception ex)
         {
-            //Will have throw a NotFoundException so need to add
-            CreateUserAccountCommand command = new(userAccountDto);
-            await _mediator.Send(command, cancellationToken);
+            _logger.LogError("Event Grid Receive Message - Failed to save Account");
```

Comment:
You could combine these 2 LogError's into a single call that passes the exception and your custom error message.

Reply:
Modified Log Message